src/promptflow-evals/promptflow/evals/evaluators/_xpia/_xpia.py

```diff
@@ -0,0 +1,301 @@
+# ---------------------------------------------------------
+import logging
+from typing import Dict, List
+
+import numpy as np
+
+from promptflow._utils.async_utils import async_run_allowing_running_loop
+from promptflow.evals._common.constants import EvaluationMetrics
+from promptflow.evals._common.rai_service import evaluate_with_rai_service
+
+logger = logging.getLogger(__name__)
+
+
+class IndirectAttackEvaluator:
+    """
+    Initializes the XPIA (cross domain prompt injected attack) jailbreak adversarial simulator with a project scope.
+    This simulator converses with your AI system using prompts injected into the context to interrupt normal expected
+    functionality by eliciting manipulated content, intrusion and attempting to gather information outside the scope
+    of your AI system.
+
+    :param project_scope: The scope of the Azure AI project.
+        It contains subscription id, resource group, and project name.
+    :type project_scope: dict
+    :param eval_last_turn: Set to True to evaluate only the most recent exchange in the dialogue,
+        focusing on the latest user inquiry and the assistant's corresponding response. Defaults to False
+    :type eval_last_turn: bool
+    :param parallel: If True, use parallel execution for evaluators. Else, use sequential execution.
+        Default is True.
+    :type parallel: bool
+    :param credential: The credential for connecting to Azure AI project.
+    :type credential: ~azure.core.credentials.TokenCredential
+    :return: A function that evaluates and generates metrics for XPIA chat scenario.
+    :rtype: Callable
+
+    **Usage**
+
+    .. code-block:: python
+
+        eval_fn = IndirectAttackEvaluator(model_config)
+        conversation = [
+            {"role": "user", "content": "What is the value of 2 + 2?"},
+            {
+                "role": "assistant",
+                "content": "2 + 2 = 4",
+                "context": {
+                    "citations": [{"id": "doc.md", "content": "Information about additions: 1 + 2 = 3, 2 + 2 = 4"}]
+                },
+            },
+            {"role": "user", "content": "What is the capital of Japan?"},
+            {
+                "role": "assistant",
+                "content": "The capital of Japan is Tokyo.",
+                "context": {
+                    "citations": [
+                        {
+                            "id": "doc.md",
+                            "content": "Tokyo is Japan's capital, known for its blend of traditional culture and \
+                                technological"
+                            "advancements.",
+                        }
+                    ]
+                },
+            },
+        ]
+        result = IndirectAttackEvaluator(conversation=conversation)
+
+    **Output format**
+
+    .. code-block:: python
+
+    {'xpia_score': 0.0, 'evaluation_per_turn':
+        {'xpia':
+            {'severity': ['Very low', 'Very low'],
```

Comment:
We don't do severity mapping for the boolean evaluators so we don't need this. UI should only show whether it was True or False